File: spring-api/src/main/java/com/datatecsolution/eprofe/spring_api/model/Alumno.java
package com.datatecsolution.eprofe.spring_api.model;

import jakarta.persistence.*;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.AllArgsConstructor;
import java.time.LocalDateTime;

import lombok.AllArgsConstructor;
import java.time.LocalDateTime;
import com.fasterxml.jackson.annotation.JsonIgnoreProperties;

@Entity
@Table(name = "alumnos")
@Data
@NoArgsConstructor
@AllArgsConstructor
@JsonIgnoreProperties({ "hibernateLazyInitializer", "handler" })
public class Alumno {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String rne;
    private String username;
    private String nombre;
    private String apellido;
    private Integer genero;

    @Column(unique = true)
    private String email;

    private String password;

    private Integer telefono;

    @Column(name = "created_at")
    private LocalDateTime createdAt;

    @Column(name = "updated_at")
    private LocalDateTime updatedAt;

    @PrePersist
    protected void onCreate() {
        createdAt = LocalDateTime.now();
        updatedAt = LocalDateTime.now();
    }

    @PreUpdate
    protected void onUpdate() {
        updatedAt = LocalDateTime.now();
    }
}

File: mobile-app/model/Alumno.ts
import { Model } from '@nozbe/watermelondb'
import { field, date, readonly } from '@nozbe/watermelondb/decorators'

export default class Alumno extends Model {
    static table = 'alumnos'

    @field('rne') rne: string
    @field('nombre') nombre: string
    @field('apellido') apellido: string
    @field('genero') genero: number
    @field('fecha_nacimiento') fechaNacimiento: number
    @field('telefono') telefono: number

    @readonly @date('created_at') createdAt: number
    @readonly @date('updated_at') updatedAt: number
}
